src/context/WalletProvider/WalletViewsSwitch.tsx

```diff
@@ -28,13 +29,18 @@ export const WalletViewsSwitch = () => {
     dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
   }
 
-  const handleBack = () => {
+  const handleBack = async () => {
     history.goBack()
     // If we're back at the select wallet modal, remove the initial route
     // otherwise clicking the button for the same wallet doesn't do anything
     if (history.location.pathname === '/') {
       dispatch({ type: WalletActions.SET_INITIAL_ROUTE, payload: '' })
     }
+
+    // If we are interacting with a KeepKey, cancel any active requests on back
+    if (state.wallet && isKeepKey(state.wallet)) {
+      await state.wallet.cancel()
```

Comment:
possible `UnhandledPromiseRejection` error

Reply:
Handled in https://github.com/shapeshift/web/pull/1341/commits/f9fe9b56dee0d189360272b2cc4ae6e9a2d20f34